src/app/ui-library/button.tsx

```diff
@@ -0,0 +1,202 @@
+
+  ${(props) => {
+    if (props.$disabled || props.$loading) {
+      switch (props.$variant) {
+        case 'primary':
+          return `
+            background-color: ${props.theme.colors.white_600};
+            color: ${props.theme.colors.elevation0};
+          `;
+        case 'secondary':
+          return `
+            border: 1px solid ${props.theme.colors.white_900};
+            color: ${props.theme.colors.white_400};
+          `;
+        case 'tertiary':
+          return `
+            color: ${props.theme.colors.white_400};
+          `;
+        case 'danger':
+          return `
+            background-color: ${props.theme.colors.danger_dark_400};
+            color: ${props.theme.colors.white_400};
+          `;
+        default:
+          return '';
+      }
+    }
+
+    switch (props.$variant) {
+      case 'primary':
+        return `
+          background-color: ${props.theme.colors.white_0};
+          color: ${props.theme.colors.elevation0};
+        `;
+      case 'secondary':
+        return `
+          border: 1px solid ${props.theme.colors.white_800};
+          color: ${props.theme.colors.white};
+        `;
+      case 'tertiary':
+        return `
+          color: ${props.theme.colors.white};
+        `;
+      case 'danger':
+        return `
+          background-color: ${props.theme.colors.danger_dark};
+          color: ${props.theme.colors.white};
+        `;
+      default:
+        return '';
+    }
+  }}
+
+  &:focus, &:hover {
+    ${(props) => {
+      if (props.$disabled) return '';
+
+      switch (props.$variant) {
+        case 'primary':
+          return `
+            background-color: ${props.theme.colors.white_200};
+          `;
+        case 'secondary':
+          return `
+            border: 1px solid ${props.theme.colors.white_850};
+            background-color: ${props.theme.colors.elevation6_800};
+          `;
+        case 'tertiary':
+          return `
+            color: ${props.theme.colors.white_200};
+          `;
+        case 'danger':
+          return `
+            background-color: ${props.theme.colors.danger_dark_100};
+          `;
+        default:
+          return '';
+      }
+    }}
+  }
+
+  &:active {
+    ${(props) => {
+      if (props.$disabled) return '';
+
+      switch (props.$variant) {
+        case 'primary':
+          return `
+            background-color: ${props.theme.colors.white_400};
+          `;
+        case 'secondary':
+          return `
+            border: 1px solid ${props.theme.colors.white};
+            background-color: ${props.theme.colors.elevation6_600};
+          `;
+        case 'tertiary':
+          return `
+            color: ${props.theme.colors.white_400};
+          `;
+        case 'danger':
+          return `
+            background-color: ${props.theme.colors.danger_dark_200};
+          `;
+        default:
+          return '';
+      }
+    }}
+  }
+`;
```

Comment:
can I have a go at rewriting this using css classes for the variants and the disabled attribute selector instead of $disabled?

Reply:
Yes please 😄

Reply:
https://github.com/secretkeylabs/xverse-web-extension/pull/766/commits/5a56cd86cec1a211157ec62dfdab2bae504d9302